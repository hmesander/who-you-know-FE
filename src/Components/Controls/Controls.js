import React from 'react';

const Controls = (props) => {
  return (
    <div className='controls-component'>
      <button className='sort-cards-button login-button' onClick={props.handleSort}>Sort Cards</button>
      <button className='play-deck-button login-button' onClick={props.handlePlay}>Play a Deck</button>
    </div>
  );
};

export default Controls;