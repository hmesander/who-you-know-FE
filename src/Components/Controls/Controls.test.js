/*eslint-disable no-undef*/

import React from 'react';
import Controls from './Controls';
import { shallow } from 'enzyme';


describe('CONTROLS TESTS', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Controls />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleSort when the sort cards button is clicked', () => {
    const mockHandleSort = jest.fn();
    const wrapper = shallow(<Controls handleSort={mockHandleSort}/>);
    const button = wrapper.find('.sort-cards-button');
    button.simulate('click');

    expect(mockHandleSort).toHaveBeenCalled();
  });

  it('should call handlePlay when the play cards button is clicked', () => {
    const mockHandlePlay = jest.fn();
    const wrapper = shallow(<Controls handlePlay={mockHandlePlay} />);
    const button = wrapper.find('.play-deck-button');
    button.simulate('click');

    expect(mockHandlePlay).toHaveBeenCalled();
  });
});