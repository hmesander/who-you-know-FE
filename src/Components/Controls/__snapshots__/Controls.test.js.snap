// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTROLS TESTS should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="sort-cards-button buttons"
          onClick={undefined}
>
          Sort Cards
</button>,
        <button
          className="play-deck-button buttons"
          onClick={undefined}
>
          Play a Deck
</button>,
      ],
      "className": "controls-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sort Cards",
          "className": "sort-cards-button buttons",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Sort Cards",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Play a Deck",
          "className": "play-deck-button buttons",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Play a Deck",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="sort-cards-button buttons"
            onClick={undefined}
>
            Sort Cards
</button>,
          <button
            className="play-deck-button buttons"
            onClick={undefined}
>
            Play a Deck
</button>,
        ],
        "className": "controls-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sort Cards",
            "className": "sort-cards-button buttons",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sort Cards",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Play a Deck",
            "className": "play-deck-button buttons",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Play a Deck",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
